### ISSUE TITLE
Support ERC-1046 by default

### ISSUE THREAD

**Issue by Pandapip1** [OpenZeppelin/openzeppelin-contracts#4149]
**🧐 Motivation**

[ERC-1046](https://eips.ethereum.org/EIPS/eip-1046) allows for ERC-20, ERC-721, and ERC-1155 to use a single method, `tokenURI` for token URIs, and adds `tokenURI` support to ERC-20. It's required to add ERC-721 and ERC-1155 tokens via [`wallet_watchAsset`](https://eips.ethereum.org/EIPS/eip-747), or to add images to ERC-20 tokens using that same RPC method.

**📝 Details**

- `tokenURI` should be added to the base ERC-1155 contract and the base ERC-20 contract
- The `name`, `symbol`, and `decimals` fields should be deprecated in ERC-20, ERC-721, and ERC-1155 (if applicable to those contracts)
- The constructors should take only a single argument, a string URL.

**Comment by treeder** [comment]
I don't think this would be a breaking change, just leave everything add `tokenURI()` and `setTokenURI(s)` to the contracts.
